src/modeler/AutoRest.Swagger/Validation/Core/RuleContext.cs

```diff
@@ -19,12 +19,16 @@ public class RuleContext
         /// Initializes a top level context for rules
         /// </summary>
         /// <param name="root"></param>
-        public RuleContext(ServiceDefinition root, Uri file) : this(null)
+        public RuleContext(ServiceDefinition root, Uri file, ServiceDefinitionMetadata metadata) : this(null)
         {
             this.Root = root;
             this.Value = root;
             this.File = file;
-            PopulateResourceTypes(root);
+            if (metadata.ServiceDefinitionDocumentType == ServiceDefinitionDocumentType.ARM)
```

Comment:
Agreed, even if it does warrant a performance investigation, I think we can revisit this once we move validator to a different repo,